src/test/java/org/padaiyal/utilities/I18nUtilityTest.java

```diff
@@ -0,0 +1,335 @@
+package org.padaiyal.utilities;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.MissingResourceException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.converter.ConvertWith;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.padaiyal.parameterconverters.ExceptionClassConverter;
+
+/**
+ * Test class for I18nUtility.
+ */
+class I18nUtilityTest {
+
+  /**
+   * The locale of the property files.
+   */
+  private static Locale locale = null;
+  /**
+   * The name of the default resource bundle.
+   */
+  private static String defaultResourceBundle = null;
+
+  /**
+   * Retrieves the required information from test.properties file.
+   *
+   * @throws IOException When there is an issue loading the properties file.
+   */
+  @BeforeAll
+  static void classSetUp() throws IOException {
+
+    PropertyUtility.addPropertyFile(I18nUtilityTest.class, "test.properties");
+    String language = PropertyUtility.getProperty("common.language");
+    String region = PropertyUtility.getProperty("common.region");
+    locale = new Locale(language, region);
+    defaultResourceBundle = PropertyUtility.getProperty("I18NUtilityTest.resourcebundle.default");
+    I18nUtility.addResourceBundle(I18nUtilityTest.class, defaultResourceBundle, locale);
+  }
+
+  /**
+   * Removes all added resource bundle except for the default resource bundle.
+   */
+  @AfterEach
+  void testTearDown() {
```

Comment:
```suggestion
  void tearDown() {
```
The `test` prefix should only be used for actual tests. Not setting up or tearing down dependencies.

Reply:
Why is the setUp() set to @BeforeAll, but the teardown is set to @AfterEach ?

Reply:
@BeforeAll is used to setup before the tests in I18nUtilityTests, but @AfterEach just cleans up after each test has ran.